platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/SocketFactory.java

```diff
@@ -62,7 +62,15 @@ static void configureAndBind(
             // set the IP_TOS option
             serverSocket.setOption(java.net.StandardSocketOptions.IP_TOS, socketConfig.ipTos());
         }
-        final InetSocketAddress endpoint = new InetSocketAddress(InetAddress.getByAddress(ALL_INTERFACES), port);
+
+        // Determine interface to bind to.
+        final InetAddress bindInterface = socketConfig.bindInterfaceHostname().equals(SocketConfig.ALL_IPV4_INTERFACES)
+                ? InetAddress.getByAddress(ALL_INTERFACES)
+                : InetAddress.getByName(socketConfig.bindInterfaceHostname());
```

Comment:
Minor: this may trigger a DNS lookup, which itself may fail, but at the very least would take extra time. Also, a single host name may resolve into multiple different IP addresses depending on the DNS configuration (e.g. via round-robin for load-balancing or similar.) Neither of this is useful or desirable when trying to bind to a particular network interface on a specific node.

I think that a more robust solution would be to require the `bindInterfaceHostname` to be an IP address (either in a string, or a byte array - the specific format is unimportant) so that:
1. Using it when opening a server socket wouldn't require extra network interactions.
2. The interface to bind to is fully deterministic.